### PR TITLE
[9.0] [MIG] account_balance_line module

### PR DESCRIPTION
**Pull request by cubells**
- Updated README
- Updated views
- Updated file structure

cc @Tecnativa

### CODE REVIEW

File: account_balance_line/README.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+or filtered by account.
+
+Usage
+=====
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/92/9.0
+
+For further information, please visit:
```
Comment (pedrobaeza) [inline account_balance_line/README.rst]: This is not needed

File: account_balance_line/__openerp__.py [hunk 2]
```diff
@@ -1,50 +1,23 @@
...
- 'license': 'AGPL-3',
- 'application': False,
- }
+# Copyright 2010-2014 Camptocamp - Vincent Renaville
+# Copyright 2016 Tecnativa - Vicent Cubells
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    'name': 'Balance on lines',
+    'summary': 'Display balance totals in move line view',
+    'version': '9.0.0.1.0',
```
Comment (pedrobaeza) [inline account_balance_line/__openerp__.py]: 9.0.1.0.0

File: account_balance_line/i18n/es.po [hunk 3]
```diff
@@ -20,12 +21,12 @@ msgstr ""
 #. module: account_balance_line
 #: view:account.move.line:account_balance_line.view_account_move_line_filter_balance
 msgid "Account"
-msgstr ""
+msgstr "Cuenta"
 
 #. module: account_balance_line
 #: field:account.move.line,line_balance:0
```
Comment (pedrobaeza) [inline account_balance_line/i18n/es.po]: This translation is not valid for version 9. Please export from v9 and translate again.

File: account_balance_line/README.rst [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+====================
+Account Balance Line
+====================
+
+Add a balance total for grouped lines in move line view.
+
+Balance field will only be shown when move lines are grouped by account
+or filtered by account.
+
+Usage
```
Comment (pedrobaeza) [inline account_balance_line/README.rst]: Add some usage instructions saying something like 
```
#. If you're an accounting adviser, enable developer mode and go to Accounting > Adviser > Journal items.
#. Or open an account or a partner, and click on Action > Journal items.
#. Filter or group by account.
#. You will see a column named "Balance" with the balance of the line.
```

File: account_balance_line/README.rst [hunk 5]
```diff
@@ -0,0 +1,56 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+====================
+Account Balance Line
+====================
+
+Add a balance total for grouped lines in move line view.
```
Comment (pedrobaeza) [inline account_balance_line/README.rst]: s/Add/This module adds

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
I have tried on runbot and I'm afraid is not working.

File: account_balance_line/README.rst [hunk 6]
```diff
@@ -6,22 +6,23 @@
...
-Add a balance total for grouped lines in move line view.
+This module adds a balance total for grouped lines in move line view.
 
 Balance field will only be shown when move lines are grouped by account
 or filtered by account.
 
 Usage
 =====
 
+If you're an accounting adviser, enable developer mode and go to
```
Comment (pedrobaeza) [inline account_balance_line/README.rst]: Put the markers I have told you in the comment: `#.`. They serve for making an unordered bullet list in RST (like `<ul>` in HTML)

**Comment by pedrobaeza** [comment]
I'm doing the last comment by myself for not disturbing you more.

**Comment by pedrobaeza** [comment]
Curious that the module doesn't get installed on runbot. I have to install it manually by myself. @moylop260, do you have any idea why this happens?

**Review by pedrobaeza — APPROVED** [review]

**Comment by moylop260** [comment]
> Curious that the module doesn't get installed on runbot. I have to install it manually by myself. @moylop260, do you have any idea why this happens?

I saw that this project has enabled [UNIT_TEST="1"](https://github.com/oca/account-financial-tools/blob/9.0/.travis.yml#L20)
This feature test all modules but delete the previous database, then we will have the database with the latest module tested.

**Comment by pedrobaeza** [comment]
Rebased for applying runbot patch in this PR

**Comment by cubells** [comment]
@pedrobaeza Thanks Pedro.

**Comment by jjscarafia** [comment]
Hi @cubells Why to create a new field and not use current v9 balance field? (I think it was not present on v8 or at least it was not stored).
Check [field definition on v9](https://github.com/odoo/odoo/blob/9.0/addons/account/models/account_move.py#L368) and [computed method](https://github.com/odoo/odoo/blob/9.0/addons/account/models/account_move.py#L309) They are the same as "line_balance" defined in this module as far as I can see.

Perhups only add "balance" in views would be enough.

**Comment by pedrobaeza** [comment]
Indeed the compute method is the same. That's inherited from v8, where this computed field doesn't exist. Then the best is to just to show the field.

File: account_balance_line/views/account_move_line_view.xml [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+        <field name="name">Journal Items add visibilty for balance</field>
+        <field name="model">account.move.line</field>
+        <field name="inherit_id" ref="account.view_account_move_line_filter"/>
+        <field name="priority" eval="9999"/>
+        <field name="arch" type="xml">
+            <field name="account_id" position="attributes">
+                <attribute name="context">{'invisible_balance': False}</attribute>
+            </field>
+            <filter context="{'group_by':'account_id'}" position="attributes">
+                <attribute name="context">{'group_by': 'account_id', 'invisible_balance': False}</attribute>
```
Comment (moylop260) [inline account_balance_line/views/account_move_line_view.xml]: same case

File: account_balance_line/views/account_move_line_view.xml [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+<odoo>
+
+    <record id="view_account_move_line_filter_balance" model="ir.ui.view">
+        <field name="name">Journal Items add visibilty for balance</field>
+        <field name="model">account.move.line</field>
+        <field name="inherit_id" ref="account.view_account_move_line_filter"/>
+        <field name="priority" eval="9999"/>
+        <field name="arch" type="xml">
+            <field name="account_id" position="attributes">
+                <attribute name="context">{'invisible_balance': False}</attribute>
```
Comment (moylop260) [inline account_balance_line/views/account_move_line_view.xml]: Use `translation="off"` more info https://github.com/odoo/odoo/issues/14497

File: account_balance_line/i18n/pt_PT.po [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+msgstr "Item do Diário"
+
+#. module: account_balance_line
+#: view:account.move.line:account_balance_line.account_move_line_balance_custom
+msgid "Total Balance"
+msgstr ""
+
+#. module: account_balance_line
+#: view:account.move.line:account_balance_line.view_account_move_line_filter_balance
+msgid "{'group_by': 'account_id', 'invisible_balance': False}"
```
Comment (moylop260) [inline account_balance_line/i18n/pt_PT.po]: Remove this translation after: `translation="off"`

File: account_balance_line/i18n/pt_PT.po [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+msgstr ""
+
+#. module: account_balance_line
+#: view:account.move.line:account_balance_line.view_account_move_line_filter_balance
+msgid "{'invisible_balance': False}"
```
Comment (moylop260) [inline account_balance_line/i18n/pt_PT.po]: same case

**Review by moylop260 — APPROVED** [review]
Small translation comment
After that LGTM

**Comment by cubells** [comment]
Done @moylop260

**Review by damendieta — APPROVED** [review]

**Pull request merged**